Sources/MessageDraft/MessageDraftChangeListener.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  MessageDraftStateListener.swift
+//
+//  Copyright (c) PubNub Inc.
+//  All rights reserved.
+//
+//  This source code is licensed under the license found in the
+//  LICENSE file in the root directory of this source tree.
+//
+
+import Foundation
+import PubNubChat
+
+/// A listener that can be used with ``MessageDraft/addChangeListener(_:)`` to listen for changes to the message draft
+/// text and get current mention suggestions.
+public protocol MessageDraftChangeListener: AnyObject {
+
+  /// Called when there is a change in the message elements or suggested mentions.
+  ///
+  /// - Parameters:
+  ///   - messageElements: An array of `MessageElement` representing current elements within the message
+  ///   - suggestedMentions: A  future object that will return the result of suggested mentions after calling its ``SuggestedMentionsFuture/async(completion:)`` method
+  func onChange(messageElements: [MessageElement], suggestedMentions: SuggestedMentionsFuture)
+}
+
+/// A closure-based implementation of the ``MessageDraftChangeListener`` protocol.
+///
+/// This class allows you to handle delegate events by passing a closure, reducing the need to implement the ``MessageDraftChangeListener`` protocol.
+/// This is useful when you want to quickly handle messages without writing additional boilerplate code.
+final public class ClosureMessageDraftChangeListener: MessageDraftChangeListener {
+  let onChangeClosure: (([MessageElement], SuggestedMentionsFuture) -> Void)
+
+  init(onChange: @escaping ([MessageElement], SuggestedMentionsFuture) -> Void) {
+    self.onChangeClosure = onChange
+  }
+
+  public func onChange(messageElements: [MessageElement], suggestedMentions: SuggestedMentionsFuture) {
+    onChangeClosure(messageElements, suggestedMentions)
+  }
+}
+
+/// A protocol representing a `[SuggestedMention]` value  that will be available in the future.
+public protocol SuggestedMentionsFuture {
```

Comment:
The concrete implementation of this protocol wraps `Future` object from `kmp-chat`. Take a look at `SuggestedMentionsFutureImpl` below.

I still wonder about possible better alternatives for that. What I wanted to definitely avoid is to not expose any symbols from `kmp-chat` dependency. I'll let you know if I come up with something better